src/pyubx2/ubxhelpers.py

```diff
@@ -296,7 +296,7 @@ def bytes2val(valb: bytes, att: str) -> object:
     elif atttyp(att) == "A":  # array of unsigned integers
         atts = attsiz(att)
         val = []

```

Comment:
I don't believe this change would be necessary if UPD-WRITE was defined with variable length - see other comments